tests/test_flask_pyoidc.py

```diff
@@ -674,3 +676,221 @@ def test_should_return_None_if_token_refresh_request_fails(self):
+            'introspection_endpoint': introspection_endpoint})
+        with self.app.test_request_context('/'):
+            flask.request.headers = {
+                'Authorization': 'Bearer access_token'
+            }
+            token_introspection_response = {
+                'active': True,
+                'aud': ['admin', 'user']
+            }
+            responses.add(responses.POST, introspection_endpoint,
+                          json=token_introspection_response)
+            self.assertFalse(authn.introspect_token(
+                flask.request, authn.clients[self.PROVIDER_NAME]))
+
+    @responses.activate
+    def test_introspect_token_should_return_False_if_required_scopes_not_permitted(self):
+
+        introspection_endpoint = f'{self.PROVIDER_BASEURL}/token/introspect'
+        authn = self.init_app(provider_metadata_extras={
+            'introspection_endpoint': introspection_endpoint})
+        with self.app.test_request_context('/'):
+            flask.request.headers = {
+                'Authorization': 'Bearer access_token'
+            }
+            token_introspection_response = {
+                'active': True,
+                'aud': ['admin', 'user', self.CLIENT_ID],
+                'scope': ['read', 'write']
+            }
+            responses.add(responses.POST, introspection_endpoint,
+                          json=token_introspection_response)
+            self.assertFalse(authn.introspect_token(
+                flask.request, authn.clients[self.PROVIDER_NAME],
+                scopes=['read', 'write', 'delete']))
+
+    @responses.activate
+    def test_introspect_token_should_return_introspection_result_if_valid_access_token(self):
+
+        introspection_endpoint = f'{self.PROVIDER_BASEURL}/token/introspect'
+        authn = self.init_app(provider_metadata_extras={
+            'introspection_endpoint': introspection_endpoint})
+        with self.app.test_request_context('/'):
+            flask.request.headers = {
+                'Authorization': 'Bearer access_token'
+            }
+            token_introspection_response = {
+                'active': True,
+                'aud': ['admin', 'user', self.CLIENT_ID],
+                'scope': 'read write delete',
+                'client_id': 'watchdogs'
+            }
+            responses.add(responses.POST, introspection_endpoint,
+                          json=token_introspection_response)
+            introspection_result = authn.introspect_token(
+                flask.request, authn.clients[self.PROVIDER_NAME])
+            self.assertEqual(token_introspection_response,
+                             introspection_result.to_dict())
+
+    def test_token_auth_should_abort_request_if_authorization_missing(self):
+
+        authn = self.init_app()
+        view_mock = self.get_view_mock()
+        with self.app.test_request_context('/'):
+            self.assertRaises(HTTPException, authn.token_auth(
```

Comment:
Would be good if this test checked the response code, which should be `401` when request is completely missing token in `Authorization` header.